shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java

```diff
@@ -0,0 +1,109 @@
+    private static Connection connection;
+    private static Channel channel;
+
+    public static void createQueueConnection(String userName, String password,
+                                            String virtualHost, String hostName, Integer portNumber)
+            throws IOException, TimeoutException {
+        if(connection==null) {
+            ConnectionFactory factory = new ConnectionFactory();
+            factory.setUsername(userName);
+            factory.setPassword(password);
+            factory.setVirtualHost(virtualHost);
+            factory.setHost(hostName);
+            factory.setPort(portNumber);
+
+            connection = factory.newConnection();
+        }
+        if(channel==null) {
+            channel = connection.createChannel();
+        }
+    }
+
+    public Connection createQueueConnection(String connectionString) throws
+            URISyntaxException, NoSuchAlgorithmException, KeyManagementException, IOException, TimeoutException {
+
+        if(connection==null) {
+            ConnectionFactory factory = new ConnectionFactory();
+            factory.setUri(connectionString);
+            connection = factory.newConnection();
+        }
+        if(channel==null){
+            channel = connection.createChannel();
+        }
+
+        return connection;
+    }
+
+    public void gracefulShutdown()
+            throws IOException, TimeoutException {
+        channel.close();
+        connection.close();
+    }
+
+    public void bindQueue(String exchangeName, String queueName, String routingKey, BuiltinExchangeType exchangeType,Boolean  isDurable)
+            throws IOException {
+        AMQP.Exchange.DeclareOk exchangeDeclareResponse =channel.exchangeDeclare(exchangeName, exchangeType, isDurable);
+        channel.queueDeclare(queueName, isDurable, false, false, null);
+        channel.queueBind(queueName, exchangeName, routingKey);
+        log.info(String.format("exchange %s is declared response %s with queue %s with RoutingKey %s", exchangeName, exchangeDeclareResponse.toString(),queueName, routingKey));
+    }
+
+    public void deleteQueue(Channel channel, String queueName, Boolean ifUnused, Boolean ifEmpty) throws IOException {
+        channel.queueDelete(queueName, ifUnused, ifEmpty);
+    }
+
+    public void purgeAllMessages(Channel channel, String queueName) throws IOException {
+        channel.queuePurge(queueName);
+    }
+
+    public void publishMessage(Channel channel, String exchangeName, String routingKey, String message) throws IOException {
+
+        if (message == null || message.equals("")) {
+            return;
+        }
+
+        byte[] messageBodyBytes = message.getBytes();
+        channel.basicPublish(exchangeName, routingKey,
+                MessageProperties.PERSISTENT_TEXT_PLAIN,
+                messageBodyBytes);
+    }
+
+    public GetResponse pullMessage(String queueName, Boolean autoAck) throws IOException {
```

Comment:
Where is visibility timeout and number of messages configuration ?

Reply:
checking how to apply that